designs/build_debug_artifacts.md

```diff
@@ -0,0 +1,120 @@
+SAM Build producing debuggable artifacts
+========================================
+
+What is the problem?
+--------------------
+
+`sam build` produces artifacts that are built for production use. In some langauges (usually interpreted), 
+these production artifacts are also debuggable locally. In the case of compiled languages, you usually need to compile
+the binary or artifact in a specific manner for them to be debuggable.  
+
+What will be changed?
+---------------------
+
+We will introduce a way in `sam build` to produce these debuggable artifacts for those compiled languages.
+
+Success criteria for the change
+-------------------------------
+
+1. Artifacts generated will be debuggable for DotNetCore2.1 Runtimes
```

Comment:
Is this 2.1 only, or will 2.0 and/or 1.0 be supported? [This doc](https://github.com/awslabs/aws-sam-cli/blob/develop/designs/dotnetcore-debugging.md) mentions both 2.0 and 2.1.

Reply:
Both 2.1 and 2.0 but not 1.0. I will update to make it more explicit.